beta-src/src/state/game/game-api-slice.ts

```diff
@@ -178,7 +178,11 @@ const startNewOrder = (
   state.order.orderID = orderForUnit.id;
   state.order.onTerritory = onTerritory;
   state.order.toTerritory = null;
-  delete state.order.type;
+  if (orderForUnit.type === "Retreat") {
+    state.order.type = "Retreat";
```

Comment:
type should not have a value until an order is completed. Please look at how the hold/move orders manipulate the `order` property in the global state to see how it should work.